### ISSUE TITLE
dependencies: eth-tester 0.1.0b5 has requirement cytoolz==0.8.2, but you'll have cytoolz 0.9.0.1 which is incompatible.

### ISSUE THREAD

**Issue by drandreaskrueger** [energywebfoundation/bond#6]
Error when installing dependencies (when following [these instructions](https://github.com/energywebfoundation/bond#installing-dependencies)):

```
pip install -r requirements.txt
...
Collecting cytoolz==0.9.0.1 (from -r requirements.txt (line 9))
...
Collecting eth-tester==0.1.0b5 (from -r requirements.txt (line 16))
  Using cached https://files.pythonhosted.org/packages/fc/b8/837289c5f482335ca44e8e35a90141bb4f6a9c386b5498d63f252b2d6218/eth_tester-0.1.0b5-py3-none-any.whl
...
eth-tester 0.1.0b5 has requirement cytoolz==0.8.2, but you'll have cytoolz 0.9.0.1 which is incompatible.
...
```

whole log:

```
pip install -r requirements.txt
Collecting asn1crypto==0.24.0 (from -r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/ea/cd/35485615f45f30a510576f1a56d1e0a7ad7bd8ab5ed7cdc600ef7cd06222/asn1crypto-0.24.0-py2.py3-none-any.whl
Collecting attrdict==2.0.0 (from -r requirements.txt (line 2))
Collecting bcrypt==3.1.4 (from -r requirements.txt (line 3))
  Using cached https://files.pythonhosted.org/packages/72/8d/57da727cbb4ddf0295b0665a93bd8f46fbbd2a48b76cbb0896ca7d27301f/bcrypt-3.1.4-cp35-cp35m-manylinux1_x86_64.whl
Collecting certifi==2018.1.18 (from -r requirements.txt (line 4))
  Using cached https://files.pythonhosted.org/packages/fa/53/0a5562e2b96749e99a3d55d8c7df91c9e4d8c39a9da1f1a49ac9e4f4b39f/certifi-2018.1.18-py2.py3-none-any.whl
Collecting cffi==1.11.5 (from -r requirements.txt (line 5))
  Using cached https://files.pythonhosted.org/packages/59/cc/0e1635b4951021ef35f5c92b32c865ae605fac2a19d724fb6ff99d745c81/cffi-1.11.5-cp35-cp35m-manylinux1_x86_64.whl
Collecting chardet==3.0.4 (from -r requirements.txt (line 6))
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting colorlog==3.1.0 (from -r requirements.txt (line 7))
  Using cached https://files.pythonhosted.org/packages/43/6b/2dcd0ce0b8a56d3e479c148ef8b7574bf6bf4512549062ca70fb67ea288f/colorlog-3.1.0-py2.py3-none-any.whl
Collecting cryptography==2.2.2 (from -r requirements.txt (line 8))
  Using cached https://files.pythonhosted.org/packages/fa/f4/3cde3604972dfa2b0fea85b9711948bb4fb70ab64095322aef35071bd254/cryptography-2.2.2-cp34-abi3-manylinux1_x86_64.whl
Collecting cytoolz==0.9.0.1 (from -r requirements.txt (line 9))
Collecting eth-abi==1.0.0 (from -r requirements.txt (line 10))
  Using cached https://files.pythonhosted.org/packages/ec/a4/be71d6e58c334f9d4ec869b9b5a7caa915f1dd950d6cfd828f96af5b6c45/eth_abi-1.0.0-py3-none-any.whl
Collecting eth-account==0.1.0a2 (from -r requirements.txt (line 11))
  Using cached https://files.pythonhosted.org/packages/7d/4e/46fffb1bd4e2da97cc3e11120b465b48f14480b6e1fb6dd8843d1f5912ab/eth_account-0.1.0a2-py3-none-any.whl
Collecting eth-hash==0.1.2 (from -r requirements.txt (line 12))
  Using cached https://files.pythonhosted.org/packages/01/df/03d97d75124c2f28e85546d9f4d6bc7ca79ce762363b9251d70f3725b1d5/eth_hash-0.1.2-py3-none-any.whl
Collecting eth-keyfile==0.5.1 (from -r requirements.txt (line 13))
  Using cached https://files.pythonhosted.org/packages/eb/a5/3615d100b62fbf20fe5d5c0d1d4d7326eac861d260b0cd2a36af2bf8ccb1/eth_keyfile-0.5.1-py3-none-any.whl
Collecting eth-keys==0.2.0b3 (from -r requirements.txt (line 14))
  Using cached https://files.pythonhosted.org/packages/9b/96/bec507291a2ecac65358b4485282ce02e8c0cdd30980a232791b355a1f5b/eth_keys-0.2.0b3-py3-none-any.whl
Collecting eth-rlp==0.1.0 (from -r requirements.txt (line 15))
  Using cached https://files.pythonhosted.org/packages/6f/9f/1daa5cbb1c263e9477738bd4a8d4e96f0468280f5f46c2079e7e8c506052/eth_rlp-0.1.0-py3-none-any.whl
Collecting eth-tester==0.1.0b5 (from -r requirements.txt (line 16))
  Using cached https://files.pythonhosted.org/packages/fc/b8/837289c5f482335ca44e8e35a90141bb4f6a9c386b5498d63f252b2d6218/eth_tester-0.1.0b5-py3-none-any.whl
Collecting eth-utils==1.0.2 (from -r requirements.txt (line 17))
  Using cached https://files.pythonhosted.org/packages/7a/10/22c5c6c1ec8275c59963af62d6dfaa24e7a68a33ae617e4289f6b0c889a1/eth_utils-1.0.2-py3-none-any.whl
Collecting hexbytes==0.1.0 (from -r requirements.txt (line 18))
  Using cached https://files.pythonhosted.org/packages/18/bd/21697d93ee2153c8c12253262b740f571e94341d4cee411504032fff3f56/hexbytes-0.1.0-py3-none-any.whl
Collecting idna==2.6 (from -r requirements.txt (line 19))
  Using cached https://files.pythonhosted.org/packages/27/cc/6dd9a3869f15c2edfab863b992838277279ce92663d334df9ecf5106f5c6/idna-2.6-py2.py3-none-any.whl
Collecting lru-dict==1.1.6 (from -r requirements.txt (line 20))
Collecting multihash==0.1.1 (from -r requirements.txt (line 21))
Collecting paramiko==2.4.1 (from -r requirements.txt (line 22))
  Using cached https://files.pythonhosted.org/packages/3e/db/cb7b6656e0e7387637ce850689084dc0b94b44df31cc52e5fc5c2c4fd2c1/paramiko-2.4.1-py2.py3-none-any.whl
Collecting pubnub==3.9.0 (from -r requirements.txt (line 23))
Collecting py-solc==2.1.0 (from -r requirements.txt (line 24))
  Using cached https://files.pythonhosted.org/packages/70/05/63a99a520d48d56cc5fdd6586a81a8c0d9b9fe9851d0417abff4a061c955/py_solc-2.1.0-py3-none-any.whl
Collecting pyasn1==0.4.2 (from -r requirements.txt (line 25))
  Using cached https://files.pythonhosted.org/packages/ba/fe/02e3e2ee243966b143657fb8bd6bc97595841163b6d8c26820944acaec4d/pyasn1-0.4.2-py2.py3-none-any.whl
Collecting pycparser==2.18 (from -r requirements.txt (line 26))
Collecting pycryptodome==3.6.0 (from -r requirements.txt (line 27))
  Using cached https://files.pythonhosted.org/packages/0c/e1/f0650ba78a913b8d1271cf24d4e0e47e1a7e32cccdaa1dfccdfa18c226e7/pycryptodome-3.6.0-cp35-cp35m-manylinux1_x86_64.whl
Collecting pycryptodomex==3.5.1 (from -r requirements.txt (line 28))
  Using cached https://files.pythonhosted.org/packages/92/66/da40b654fd4b2518022982f6ad99ab01246c8225dc931d68271859d29385/pycryptodomex-3.5.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting PyJWT==1.6.1 (from -r requirements.txt (line 29))
  Using cached https://files.pythonhosted.org/packages/31/8f/19c302aa9a391dd1fbd249362b749021b88d40fb59af0363939a2250afed/PyJWT-1.6.1-py2.py3-none-any.whl
Collecting pylru==1.0.9 (from -r requirements.txt (line 30))
Collecting PyNaCl==1.2.1 (from -r requirements.txt (line 31))
  Using cached https://files.pythonhosted.org/packages/c9/9c/c8d4ae0f590faded4af55381131ad4db8729214eba6537c520622ba3548a/PyNaCl-1.2.1-cp35-cp35m-manylinux1_x86_64.whl
Collecting pyotp==2.2.6 (from -r requirements.txt (line 32))
  Using cached https://files.pythonhosted.org/packages/12/1f/0b27e24456a9bb89b418602f61cbcad4861b55cf872469643ed13b38ff9b/pyotp-2.2.6-py2.py3-none-any.whl
Collecting pysha3==1.0.2 (from -r requirements.txt (line 33))
  Using cached https://files.pythonhosted.org/packages/4b/a5/7eb00630fa4dc9751464faab8b5908706a4190a3ab2a37b2c03cabb2c9a8/pysha3-1.0.2-cp35-cp35m-manylinux1_x86_64.whl
Collecting requests==2.18.4 (from -r requirements.txt (line 34))
  Using cached https://files.pythonhosted.org/packages/49/df/50aa1999ab9bde74656c2919d9c0c085fd2b3775fd3eca826012bef76d8c/requests-2.18.4-py2.py3-none-any.whl
Collecting resin-sdk==2.2.0 (from -r requirements.txt (line 35))
Collecting rlp==0.6.0 (from -r requirements.txt (line 36))
Collecting semantic-version==2.6.0 (from -r requirements.txt (line 37))
  Using cached https://files.pythonhosted.org/packages/28/be/3a7241d731ba89063780279a5433f5971c1cf41735b64a9f874b7c3ff995/semantic_version-2.6.0-py3-none-any.whl
Collecting six==1.11.0 (from -r requirements.txt (line 38))
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting toolz==0.9.0 (from -r requirements.txt (line 39))
Collecting typing==3.6.2 (from -r requirements.txt (line 40))
  Using cached https://files.pythonhosted.org/packages/44/88/d09c6a7fe1af4a02f16d2f1766212bec752aadb04e5699a9706a10a1a37d/typing-3.6.2-py3-none-any.whl
Collecting urllib3==1.22 (from -r requirements.txt (line 41))
  Using cached https://files.pythonhosted.org/packages/63/cb/6965947c13a94236f6d4b8223e21beb4d576dc72e8130bd7880f600839b8/urllib3-1.22-py2.py3-none-any.whl
Collecting web3==4.1.0 (from -r requirements.txt (line 42))
  Using cached https://files.pythonhosted.org/packages/4c/a2/9f2cf0fc2c7938871586061491f7be9a2a67028dad091e01bf781195eb87/web3-4.1.0-py3-none-any.whl
Collecting websockets==4.0.1 (from -r requirements.txt (line 43))
  Using cached https://files.pythonhosted.org/packages/4a/30/f68f8e72901609397ced5cd5b59002716b3e64b1a364631124820a06fe0a/websockets-4.0.1-cp35-cp35m-manylinux1_x86_64.whl
eth-tester 0.1.0b5 has requirement cytoolz==0.8.2, but you'll have cytoolz 0.9.0.1 which is incompatible.
Installing collected packages: asn1crypto, six, attrdict, pycparser, cffi, bcrypt, certifi, chardet, colorlog, idna, cryptography, toolz, cytoolz, eth-hash, eth-utils, eth-abi, hexbytes, rlp, eth-rlp, pycryptodome, eth-keys, eth-keyfile, eth-account, semantic-version, eth-tester, lru-dict, multihash, pyasn1, PyNaCl, paramiko, pycryptodomex, urllib3, requests, pubnub, py-solc, PyJWT, pylru, pyotp, pysha3, resin-sdk, typing, websockets, web3
Successfully installed PyJWT-1.6.1 PyNaCl-1.2.1 asn1crypto-0.24.0 attrdict-2.0.0 bcrypt-3.1.4 certifi-2018.1.18 cffi-1.11.5 chardet-3.0.4 colorlog-3.1.0 cryptography-2.2.2 cytoolz-0.9.0.1 eth-abi-1.0.0 eth-account-0.1.0a2 eth-hash-0.1.2 eth-keyfile-0.5.1 eth-keys-0.2.0b3 eth-rlp-0.1.0 eth-tester-0.1.0b5 eth-utils-1.0.2 hexbytes-0.1.0 idna-2.6 lru-dict-1.1.6 multihash-0.1.1 paramiko-2.4.1 pubnub-3.9.0 py-solc-2.1.0 pyasn1-0.4.2 pycparser-2.18 pycryptodome-3.6.0 pycryptodomex-3.5.1 pylru-1.0.9 pyotp-2.2.6 pysha3-1.0.2 requests-2.18.4 resin-sdk-2.2.0 rlp-0.6.0 semantic-version-2.6.0 six-1.11.0 toolz-0.9.0 typing-3.6.2 urllib3-1.22 web3-4.1.0 websockets-4.0.1
```

**Comment by drandreaskrueger** [comment]
funny that this happens actually, because the whole purpose of a `pip install -r requirements.txt` is that me the enduser has the exact same versions with which you have tested your code. Aren't you testing your own instructions?


First attempt at fixing failed:

just changing in `requirements.txt` to `cytoolz==0.8.2` does not help, because then later this happens:

```
eth-keyfile 0.5.1 has requirement cytoolz<1.0.0,>=0.9.0, but you'll have cytoolz 0.8.2 which is incompatible.
web3 4.1.0 has requirement cytoolz<1.0.0,>=0.9.0, but you'll have cytoolz 0.8.2 which is incompatible.
eth-keys 0.2.0b3 has requirement cytoolz<1.0.0,>=0.9.0, but you'll have cytoolz 0.8.2 which is incompatible.
```

so, you please figure out a correct, and working `requirements.txt` because [yours](https://github.com/energywebfoundation/bond/blob/master/bond/requirements.txt) is wrong.

**Comment by drandreaskrueger** [comment]
Then I have tried to see what happens if I do [not give any version number](https://gist.github.com/drandreaskrueger/6a6259f9d0856535f1ed1f1585c2bc33) ...

but that also results in conflicts:

```
pip install -r requirements_unversioned.txt 

...

  Running setup.py bdist_wheel for resin-sdk ... done
  Stored in directory: ~/.cache/pip/wheels/71/31/c2/ac6ae0c7f3ba45999bec0f6bef640050449ed3962feddc987f
Successfully built pubnub pylru resin-sdk

resin-sdk 4.7.0 has requirement pubnub<4.0.0, but you'll have pubnub 4.1.0 which is incompatible.

web3 4.3.0 has requirement websockets<5.0.0,>=4.0.1, but you'll have websockets 5.0.1 which is incompatible.

...
```

which I have reported to both projects:

**Comment by drandreaskrueger** [comment]
please you figure out a correct, and working `requirements.txt` - thanks.

**Comment by cerealkill** [comment]
Hi, the problem lies in [web3.py](https://github.com/ethereum/web3.py) requirements which weren't faulty by the time of development. In my machine is just a warning tho, the software works still. Is it the same for you?

**Comment by cerealkill** [comment]
Fixed in the new `web3` verison `4.6.0`.
